lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -177,6 +220,25 @@ def get(self, request, *args, **kwargs):
 
         return Response(serializer.data)
 
+    def _get_start_block(self, blocks, block):
+        if not blocks.get_children(block):
+            return block
+
+        children = blocks.get_children(block)
+        start_block = self._get_start_block(blocks, children[0])
+        return start_block
+
+    def _get_resume_block(self, blocks, block):
+        if not blocks.get_children(block):
+            return block
+
+        children = blocks.get_children(block)
+        for child in children:
+            is_resume_block = blocks.get_xblock_field(child, 'resume_block', False)
+            if is_resume_block:
+                resume_block = self._get_resume_block(blocks, child)
+        return resume_block
```

Comment:
What about the following, which is a little more succinct (doesn't call `get_children` twice, doesn't continue loop after finding resume block, doesn't potentially return an undefined variable):
```
    def _get_resume_block(self, blocks, block):
        children = blocks.get_children(block)
        if not children:
            return block

        for child in children:
            if blocks.get_xblock_field(child, 'resume_block', False):
                return self._get_resume_block(blocks, child)
        return None
```